### PR TITLE
Add go-back button

### PR DESCRIPTION
**Pull request by vasylshpak**
Added "go-back button" on:
- product/create
- produt/edit
- category/create/edit
- users/edit
- news/add
- order/details
- news/edit

Admin has the ability to move back from all places across the Admin Panel.

image

### CODE REVIEW

**Review by dieie32 — APPROVED** [review]

File: src/components/news-form/news-form.js [hunk 1]
```diff
@@ -158,6 +159,18 @@ const NewsForm = ({ article, id }) => {
           </TabPanel>
         ))}
       </form>
+
+      <div className={classes.controlsBlock}>
+        <Link to={config.routes.pathToNews}>
+          <StandardButton
+              id='back'
+              title='Назад'
+              variant='outlined'
```
Comment (nadiyafomenko) [inline src/components/news-form/news-form.js:168]: Please move 'Назад' title to config file

File: src/components/news-form/news-form.js [hunk 2]
```diff
@@ -158,6 +159,18 @@ const NewsForm = ({ article, id }) => {
...
+              onClickHandler={() => {}}
+              data-cy='back-btn'
```
Comment (nadiyafomenko) [inline src/components/news-form/news-form.js:170]: Why this handler is empty?
You can do smth like this
        ```
  <StandardButton
            title={config.buttonTitles.BACK}
            onClickHandler={() => dispatch(push(pathToNews))}
          />
```

File: src/pages/categories/categories-add/categories-add.js [hunk 3]
```diff
@@ -517,14 +519,26 @@ const CategoriesAdd = ({ id, editMode }) => {
...
+      <div className={classes.controlsBlock}>
+        <Link to={config.routes.pathToCategories}>
+          <StandardButton
+              id='back'
+              title='Назад'
+              variant='outlined'
+              onClickHandler={() => {}}
+              data-cy='back-btn'
+          />
+        </Link>
```
Comment (nadiyafomenko) [inline src/pages/categories/categories-add/categories-add.js:532]: Please move title to config, and set link to onclickHandler

File: src/pages/news/news-add/news-add.js [hunk 4]
```diff
@@ -183,6 +184,17 @@ const NewsAdd = () => {
...
+          <div className={classes.controlsBlock}>
+            <Link to={config.routes.pathToNews}>
+              <StandardButton
+                  id='back'
+                  title='Назад'
+                  variant='outlined'
+                  onClickHandler={() => {}}
+                  data-cy='back-btn'
+              />
+            </Link>
```
Comment (nadiyafomenko) [inline src/pages/news/news-add/news-add.js:196]: Please move title to config, and set link to onclickHandler

File: src/pages/order-item/order-item.js [hunk 5]
```diff
@@ -104,6 +108,17 @@ const OrderItem = ({ id }) => {
...
+        <Link to={config.routes.pathToOrders}>
+          <StandardButton
+              id='back'
+              title='Назад'
+              variant='outlined'
+              onClickHandler={() => {}}
+              data-cy='back-btn'
+          />
+        </Link>
+      </div>
```
Comment (nadiyafomenko) [inline src/pages/order-item/order-item.js:121]: Please move title to config, and set link to onclickHandler

File: src/pages/products/product-add/products-add.js [hunk 6]
```diff
@@ -131,7 +134,19 @@ const ProductsAdd = () => {
...
+      <div className={styles.controlsBlock}>
+        <Link to={config.routes.pathToProducts}>
+          <StandardButton
+              id='back'
+              title='Назад'
+              variant='outlined'
+              onClickHandler={() => {}}
+              data-cy='back-btn'
+          />
+        </Link>
```
Comment (nadiyafomenko) [inline src/pages/products/product-add/products-add.js:146]: Please move title to config, and set link to onclickHandler

File: src/components/news-form/news-form.js (hunk 2 above)
Comment (vasylshpak) [reply src/components/news-form/news-form.js]: done

File: src/pages/news/news-add/news-add.js (hunk 4 above)
Comment (vasylshpak) [reply src/pages/news/news-add/news-add.js]: done

File: src/pages/order-item/order-item.js (hunk 5 above)
Comment (vasylshpak) [reply src/pages/order-item/order-item.js]: done

File: src/pages/products/product-add/products-add.js (hunk 6 above)
Comment (vasylshpak) [reply src/pages/products/product-add/products-add.js]: done

File: src/pages/categories/categories-add/categories-add.js (hunk 3 above)
Comment (vasylshpak) [reply src/pages/categories/categories-add/categories-add.js]: done

File: src/components/news-form/news-form.js (hunk 1 above)
Comment (vasylshpak) [reply src/components/news-form/news-form.js]: done

File: src/pages/products/product-edit/product-edit-form/product-edit-form.js [hunk 7]
```diff
@@ -303,6 +305,17 @@ const ProductEditForm = () => {
...
       </Grid>
+      <div className={styles.controlsBlock}>
+        <Link to={config.routes.pathToProducts}>
+          <StandardButton
+            id='back'
+            title={config.buttonTitles.GO_BACK_TITLE}
+            variant='outlined'
+            onClickHandler={() => {}}
+            data-cy='back-btn'
+          />
```
Comment (nadiyafomenko) [inline src/pages/products/product-edit/product-edit-form/product-edit-form.js:316]: You only forgot about this part

**Review by nadiyafomenko — COMMENTED** [review]
nice job, thanks

**Review by nadiyafomenko — APPROVED** [review]

File: src/pages/products/product-edit/product-edit-form/product-edit-form.js (hunk 7 above)
Comment (vasylshpak) [reply src/pages/products/product-edit/product-edit-form/product-edit-form.js]: thank you, fixed!

**Review by ahd3r — APPROVED** [review]

File: src/components/news-form/news-form.js [hunk 8]
```diff
@@ -158,6 +159,16 @@ const NewsForm = ({ article, id }) => {
           </TabPanel>
         ))}
       </form>
+
+      <div className={classes.controlsBlock}>
+        <StandardButton
+          id='back'
```
Comment (maksym-strus) [inline src/components/news-form/news-form.js:165]: Use the same label as data-cy attribute has

File: src/components/news-form/news-form.js [hunk 9]
```diff
@@ -158,6 +159,16 @@ const NewsForm = ({ article, id }) => {
...
+          title={config.buttonTitles.GO_BACK_TITLE}
+          variant='outlined'
+          onClickHandler={() => dispatch(push(config.routes.pathToNews))}
```
Comment (maksym-strus) [inline src/components/news-form/news-form.js:168]: Make sense to move this dispatch to a separated handler function

File: src/pages/categories/categories-add/categories-add.js [hunk 10]
```diff
@@ -517,14 +518,24 @@ const CategoriesAdd = ({ id, editMode }) => {
...
-        className={classes.saveBtn}
-        onClick={editMode ? handleCategoryEdit : handleCategorySave}
-      >
-        {config.buttonTitles.titleGenerator(editMode, isMain)}
-      </Button>
+
+      <div className={classes.controlsBlock}>
+        <StandardButton
+          id='back'
+          title={config.buttonTitles.GO_BACK_TITLE}
```
Comment (maksym-strus) [inline src/pages/categories/categories-add/categories-add.js:525]: Use the same label as data-cy attribute has

File: src/pages/categories/categories-add/categories-add.js [hunk 11]
```diff
@@ -517,14 +518,24 @@ const CategoriesAdd = ({ id, editMode }) => {
...
+          variant='outlined'
+          onClickHandler={() => dispatch(push(config.routes.pathToCategories))}
```
Comment (maksym-strus) [inline src/pages/categories/categories-add/categories-add.js:527]: Make sense to move this dispatch to a separated handler function

File: src/pages/news/news-add/news-add.js [hunk 12]
```diff
@@ -183,6 +184,15 @@ const NewsAdd = () => {
               required
             />
           </Paper>
+          <div className={classes.controlsBlock}>
+            <StandardButton
+              id='back'
```
Comment (maksym-strus) [inline src/pages/news/news-add/news-add.js:189]: Check previous comments

File: src/pages/news/news-add/news-add.js [hunk 13]
```diff
@@ -183,6 +184,15 @@ const NewsAdd = () => {
...
+              title={config.buttonTitles.GO_BACK_TITLE}
+              variant='outlined'
+              onClickHandler={() => dispatch(push(config.routes.pathToNews))}
```
Comment (maksym-strus) [inline src/pages/news/news-add/news-add.js:192]: Check previous comments

File: src/pages/order-item/order-item.js [hunk 14]
```diff
@@ -47,65 +52,101 @@ const OrderItem = ({ id }) => {
...
-    if(selectedOrder) {
+  useEffect(() => {
+    if (selectedOrder) {
       resetForm({ values: selectedOrder });
     }
-  },[selectedOrder,resetForm])
+  }, [selectedOrder, resetForm]);
 
-  const formikHandleChange = ['CREATED','CONFIRMED'].includes(selectedOrder && selectedOrder.status) ? handleChange : ()=> {}
+  const formikHandleChange = ['CREATED', 'CONFIRMED'].includes(
```
Comment (maksym-strus) [inline src/pages/order-item/order-item.js:84]: You use [...] array in 2 different places. Maybe make sense move it to separate variable

File: src/pages/order-item/order-item.js [hunk 15]
```diff
@@ -47,65 +52,101 @@ const OrderItem = ({ id }) => {
...
+          variant='contained'
+          color='primary'
+          className={classes.saveBtn}
+        >
+          {SAVE_ORDER}
+        </Button>
+      )}
+      <div className={classes.controlsBlock}>
+        <StandardButton
+          id='back'
```
Comment (maksym-strus) [inline src/pages/order-item/order-item.js:137]: Check previous comments

File: src/pages/order-item/order-item.js [hunk 16]
```diff
@@ -47,65 +52,101 @@ const OrderItem = ({ id }) => {
...
+          title={config.buttonTitles.GO_BACK_TITLE}
+          variant='outlined'
+          onClickHandler={() => dispatch(push(config.routes.pathToOrders))}
```
Comment (maksym-strus) [inline src/pages/order-item/order-item.js:140]: Check previous comments

File: src/pages/products/product-add/products-add.js [hunk 17]
```diff
@@ -131,6 +134,15 @@ const ProductsAdd = () => {
   return (
     <Paper className={styles.container}>
       <CustomizedStepper steps={steps} activeStep={activeStep} />
+      <div className={styles.controlsBlock}>
+        <StandardButton
+          id='back'
```
Comment (maksym-strus) [inline src/pages/products/product-add/products-add.js:139]: Check previous comments

File: src/pages/products/product-add/products-add.js [hunk 18]
```diff
@@ -131,6 +134,15 @@ const ProductsAdd = () => {
...
+          title={config.buttonTitles.GO_BACK_TITLE}
+          variant='outlined'
+          onClickHandler={() => dispatch(push(config.routes.pathToProducts))}
```
Comment (maksym-strus) [inline src/pages/products/product-add/products-add.js:142]: Check previous comments

File: src/pages/products/product-edit/product-edit-form/product-edit-form.js [hunk 19]
```diff
@@ -303,6 +304,15 @@ const ProductEditForm = () => {
         </Grid>
         <CommentsPage productId={product._id} />
       </Grid>
+      <div className={styles.controlsBlock}>
+          <StandardButton
+            id='back'
```
Comment (maksym-strus) [inline src/pages/products/product-edit/product-edit-form/product-edit-form.js:309]: Check previous comments

File: src/pages/products/product-edit/product-edit-form/product-edit-form.js [hunk 20]
```diff
@@ -303,6 +304,15 @@ const ProductEditForm = () => {
...
+            title={config.buttonTitles.GO_BACK_TITLE}
+            variant='outlined'
+             onClickHandler={() => dispatch(push(config.routes.pathToProducts))}
```
Comment (maksym-strus) [inline src/pages/products/product-edit/product-edit-form/product-edit-form.js:312]: Check previous comments

File: src/pages/users/users-details/users-details.js [hunk 21]
```diff
@@ -92,6 +94,16 @@ const UsersDetails = (props) => {
           buttonStatus={buttonStatus}
           buttonHandler={() => userStatusHandler(id)}
         />
+        <div className={styles.controlsBlock}>
+          <StandardButton
+            className={styles.userDetails}
+            id='back'
```
Comment (maksym-strus) [inline src/pages/users/users-details/users-details.js:100]: Check previous comments

File: src/pages/users/users-details/users-details.js [hunk 22]
```diff
@@ -92,6 +94,16 @@ const UsersDetails = (props) => {
...
+            title={config.buttonTitles.GO_BACK_TITLE}
+            variant='outlined'
+            onClickHandler={() => dispatch(push(config.routes.pathToUsers))}
```
Comment (maksym-strus) [inline src/pages/users/users-details/users-details.js:103]: Check previous comments

**Review by nadiyafomenko — APPROVED** [review]

File: src/components/news-form/news-form.js (hunk 8 above)
Comment (vasylshpak) [reply src/components/news-form/news-form.js]: done

File: src/components/news-form/news-form.js (hunk 9 above)
Comment (vasylshpak) [reply src/components/news-form/news-form.js]: done

File: src/pages/categories/categories-add/categories-add.js (hunk 10 above)
Comment (vasylshpak) [reply src/pages/categories/categories-add/categories-add.js:528]: done

File: src/pages/categories/categories-add/categories-add.js (hunk 11 above)
Comment (vasylshpak) [reply src/pages/categories/categories-add/categories-add.js]: done

File: src/pages/news/news-add/news-add.js (hunk 12 above)
Comment (vasylshpak) [reply src/pages/news/news-add/news-add.js]: done

File: src/pages/news/news-add/news-add.js (hunk 13 above)
Comment (vasylshpak) [reply src/pages/news/news-add/news-add.js]: done

File: src/pages/order-item/order-item.js (hunk 14 above)
Comment (vasylshpak) [reply src/pages/order-item/order-item.js]: done

File: src/pages/order-item/order-item.js (hunk 15 above)
Comment (vasylshpak) [reply src/pages/order-item/order-item.js]: done

File: src/pages/order-item/order-item.js (hunk 16 above)
Comment (vasylshpak) [reply src/pages/order-item/order-item.js]: done

File: src/pages/products/product-add/products-add.js (hunk 17 above)
Comment (vasylshpak) [reply src/pages/products/product-add/products-add.js]: done

File: src/pages/products/product-add/products-add.js (hunk 18 above)
Comment (vasylshpak) [reply src/pages/products/product-add/products-add.js]: done

File: src/pages/products/product-edit/product-edit-form/product-edit-form.js (hunk 19 above)
Comment (vasylshpak) [reply src/pages/products/product-edit/product-edit-form/product-edit-form.js]: done

File: src/pages/users/users-details/users-details.js (hunk 22 above)
Comment (vasylshpak) [reply src/pages/users/users-details/users-details.js]: done

File: src/pages/users/users-details/users-details.js (hunk 21 above)
Comment (vasylshpak) [reply src/pages/users/users-details/users-details.js]: done

**Pull request merged**